api/views.py

```diff
@@ -151,14 +151,24 @@ def partial_update(self, request, pk=None, **kwargs):
                 )
                 serializer.is_valid(raise_exception=True)
                 data = serializer.validated_data
+                new_url = data.get("url", puzzle.url)
+                is_new_url = new_url != puzzle.url
                 puzzle.update_metadata(
                     new_name=data.get("name", puzzle.name),
-                    new_url=data.get("url", puzzle.url),
+                    new_url=new_url,
                     new_is_meta=data.get("is_meta", puzzle.is_meta),
                 )
                 if "status" in data:
                     puzzle.status = data["status"]
                     puzzle.save()
+
+                if is_new_url and google_api_lib.enabled():
+                    google_api_lib.task.add_puzzle_link_to_sheet.delay(
+                        new_url, puzzle.sheet
+                    )
+                if puzzle.is_meta:
+                    google_api_lib.task.update_meta_sheet_feeders.delay(puzzle.id)
```

Comment:
Actually could we wrap these in `transaction.on_commit` ? If the celery tasks run before the transaction commits (or if it doesn't commit successfully) the task is going to error.